auslib/test/blobs/test_base.py

```diff
@@ -63,3 +67,132 @@ def testSchemaCaching(self):
             blob.validate('fake', [])
 
             self.assertEquals(yaml_load.call_count, 1)
+
+
+# Things we consider to be useful values in testing blobs. Basically, these
+# are things would actually show up in real blobs. Nesting is handled
+# further down.
+# Ideally we'd include st.boolean() here, but without a way to exclude
+# booleans from list items (which doesn't happen in the real world), we end
+# up with test failures because things like "1 in [True]" are True, which
+# ends up failing tests.
+useful_values = st.none() | st.floats(allow_nan=False) | st.text()
+
+
+@given(st.lists(useful_values), st.lists(useful_values), st.lists(useful_values))
+def test_merge_lists_unique_items_present(base, left_additional, right_additional):
+    left = deepcopy(base) + left_additional
+    right = deepcopy(base) + right_additional
+    expected = sorted(list(set(deepcopy(base) + left_additional + right_additional)))
+    got = sorted(merge_lists(base, left, right))
```

Comment:
[Purist mode on] The use of sorting in the test ignores the convention that lists are ordered. I don't think that actually matters for partials/completes down at the locale level, but could you go
   ``expected = deepcopy(base) + left_additional + [l for l in right_additional if l not in left_additional]``
and make the test stronger ?

Reply:
Good idea, let's do that.

In an ideal world, we'd be treating partials/completes as sets (or maybe dicts) everywhere and this wouldn't be an issue.